src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -2572,7 +2808,7 @@ FixedwingPositionControl::initializeAutoLanding(const hrt_abstime &now, const po
 			// TODO: proper handling of on-the-fly landing points would need to involve some more sophisticated
 			// landing pattern generation and corresponding logic
 
-			height_above_land_point = _current_altitude - pos_sp_curr.alt;
+			height_above_land_point = _current_altitude - land_point_altitdue;
 			local_approach_entrance = local_position;
 		}
```

Comment:
maybe we could move this conditional outside of this method, and just input an "approach" altitude instead of the entire previous waypoint struct? then outside just determine whether to put in current altitude or last waypoint altitude. -- then the method doesnt need to understand anything about previous waypoint validity.

Reply:
But the same also would need to be done for local_approach_entrance, right? As it also consumes the previous wp. Not that sure if it then really is cleaner.

Reply:
ah missed that. sure. though i did spot one more thing now `land_point_altitdue` < typo

Reply:
Thanks, fixed it.